functional/streams.py

```diff
@@ -47,6 +50,17 @@ def __call__(self, *args, no_wrap=None, **kwargs):
             args, engine, no_wrap=default_value(no_wrap, self.no_wrap, False)
         )
 
+    def chain(self, *args, no_wrap=None, **kwargs):
```

Comment:
self should be cls (and use @classmethod) here right? (looking at your tests)

Reply:
> self should be cls (and use @classmethod) here right? (looking at your tests)

I don't think so. Such as `range`, it should be classmethod too ?